admin_manual/configuration/performance_tips.rst

```diff
@@ -43,3 +43,173 @@ Memcached servers must be specified in the "memcached_servers" array in ownCloud
+   | ``sudo dpkg-buildpackage -uc -b``
+   | ``ls -la /opt``
+   | ``sudo dpkg --install /opt/nginx_{release}~{distribution}_amd64.deb``
+   |
+
+5. | *Check if the compilation and installation of the* ``ngx_cache_purge`` *module was successful*
+   | ``nginx -V 2>&1 | grep  ngx_cache_purge -o``
+   | 
+   | It should show now: ``ngx_cache_purge``
+   | 
+   | Show Nginx version including all features compiled and installed:
+   | ``nginx -V 2>&1 | sed s/" --"/"\n\t--"/g``
+   |
+
+6. | *Mark Nginx to be blocked from further updates via* ``apt-get``
+   | ``sudo dpkg --get-selections | grep nginx``
+   | 
+   | For eyery nginx component listed do a:
+   | ``sudo apt-mark hold <component>``
+   |
+
+7. | *Regular checks for nginx updates*
+   | Do a regular visit on the `Nginx news page <http://nginx.org>`_ and proceed in case of updates with item 2 to 5
+
+Configure Nginx with the ``nginx-cache-purge`` module
+-----------------------------------------------------
+1. | *Preperation*
+   | Create a directory where Nginx will save the cached thumbnails. Use any path that fits to your environment. Replace ``{path}`` with the path used, example path below:
+   | ``sudo mkdir -p /usr/local/tmp/cache``
+   |
+
+2. | *Configuration*
+   | ``sudo vi /etc/nginx/sites-enabled/{your-owncloud-nginx-config-file}``
+   | 
+   | Note: the ``keys_zone`` / ``fastcgi_cache`` name and the ``{path}`` must be unique to each instance of owncloud serverd with Nginx !
+   | 
+   | Add at the *beginning*, but *outside* the ``server{}`` block:
+   | ``fastcgi_cache_path {path} levels=1:2 keys_zone=OWNCLOUD:100m inactive=60m;``
+   |
+   | Add *inside* the ``server{}`` block, as an example of a configuration:
+   |
+   | ``set $skip_cache 1;``
+   |    
+   | ``# POST requests and urls with a query string should always go to PHP``
+   | ``if ($request_uri ~* "thumbnail.php") {``
+   |      ``set $skip_cache 0;``
+   |   ``}``
+   |    
+   |  ``fastcgi_cache_key "$scheme$request_method$host$request_uri";``
+   |  ``fastcgi_cache_use_stale error timeout invalid_header http_500;``
+   |  ``fastcgi_ignore_headers Cache-Control Expires Set-Cookie;``
+   |    
+   |  ``location ~ \.php(?:$|/) {``
+   |      ``fastcgi_split_path_info ^(.+\.php)(/.+)$;``
+   |      ``try_files $fastcgi_script_name =404;``
+   |    
+   |      ``include fastcgi_params;``
+   |      ``fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;``
+   |      ``fastcgi_param PATH_INFO $fastcgi_path_info;``
+   |      ``fastcgi_param HTTPS on;``
+   |      ``fastcgi_pass php-handler;``
+   |    
+   |      ``fastcgi_cache_bypass $skip_cache;``
+   |      ``fastcgi_no_cache $skip_cache;``
+   |      ``fastcgi_cache OWNCLOUD;``
+   |      ``fastcgi_cache_valid  60m;``
+   |    ``}``
+   |
+   | Note regarding the ``fastcgi_pass`` parameter:
+   | Use whatever fits your configuration. In the example above, a ``upstream`` was defined in a global configuration file.
+   | This then can look like:
```

Comment:
Regarding the global configuration file mentioned:

This is part of any nginx configuration, especially when using fastcgi, and is not necessary to be explained here. The reason I put it there was, because there is the parameter: ``fastcgi_pass php-handler;`` The reference is ``php-handler`` which needs a conterpart which is in the ``upstream`` definition. So those who configure nginx know now given relationship.

For your info: the term ``php-handler`` can be any term, but it must have his correspondent counterpart in a definition how to access fastcgi.